src/test/java/core/basesyntax/StrategyTest.java

```diff
@@ -0,0 +1,95 @@
+package core.basesyntax;
+
+import java.lang.reflect.Method;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class StrategyTest {
+  private static final double DELTA = 0.0;
+  private static List<Double> discounts;
+  private static List<String> specialEvents;
+
+  @BeforeClass
+  public static void setUp() {
+    specialEvents = List.of("Birthday", "Black Friday", "New Year", "Other event");
+    discounts = List.of(33.0, 45.0, 20.0, 0.0);
+  }
+
+  @Test
+  public void getDiscountBySpecialEvent_IsOk() {
+    for (int i = 0; i < discounts.size(); i++) {
+      DiscountCenter discountCenter = new DiscountCenter();
+      double expectedDiscount = discounts.get(i);
+      double actualDiscount = discountCenter.getDiscountBySpecialEvent(specialEvents.get(i));
+      Assert.assertEquals("Expected discount " + expectedDiscount
+              + ",but your discount " + actualDiscount,
+          expectedDiscount, actualDiscount, DELTA);
+    }
+  }
+
+  @Test
+  public void interfaceDiscountService_IsCreated() {
+    try {
+      Class<?> discountStrategyInterface = Class.forName("core.basesyntax.strategy.DiscountService");
```

Comment:
Which class is the direct implementation of strategy pattern? `DiscountService`, `DiscountCenter` or `BirthdayDiscountService`?